src/DistributeShifts.h

```diff
@@ -0,0 +1,23 @@
+#ifndef HALIDE_DISTRIBUTE_SHIFTS_H
+#define HALIDE_DISTRIBUTE_SHIFTS_H
+
+/** \file
+ * A tool to distribute shifts as multiplies, useful for some backends. (e.g. ARM, HVX).
+ */
+
+#include "IR.h"
+
+namespace Halide {
+namespace Internal {
+
+// Distributes shifts as multiplies. If `polynomials_only` is set,
+// then only distributes the patterns `a + widening_shl(b, c)` /
```

Comment:
I'm a bit confused by the term polynomial here too. I don't see where the polynomial is. Is there a better word?

Reply:
Perhaps. I’m trying to say that a + b << c is lowered to a + (b * (1 << c)) but just b << c will not be (on ARM)

Reply:
Do you know of a better word? “affine” perhaps?

Reply:
multiply_add?